osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -277,13 +280,18 @@ private boolean checkSsldPendingFirmwareUpdate(final CorrelationIds ids,
 
     private void checkFirmwareHistory(final String deviceId,
             final List<org.opensmartgridplatform.domain.core.valueobjects.FirmwareVersion> firmwareVersions) {
-        final List<org.opensmartgridplatform.domain.core.valueobjects.FirmwareVersion> versionsNotInHistory = this
-                .checkFirmwareHistoryForVersion(deviceId, firmwareVersions);
-        for (final org.opensmartgridplatform.domain.core.valueobjects.FirmwareVersion firmwareVersion : versionsNotInHistory) {
-            LOGGER.info("Firmware version {} is not in history of device {}, we are trying to add it", firmwareVersion,
-                    deviceId);
-            this.tryToAddFirmwareVersionToHistory(deviceId, firmwareVersion);
-        }
+        final List<org.opensmartgridplatform.domain.core.valueobjects.FirmwareVersion> firmwareVersionsNotCurrent = this
+                .checkFirmwareHistoryForModuleVersionsNotCurrentlyInstalled(deviceId, firmwareVersions);
+        this.tryToAddFirmwareVersionToHistory(deviceId, firmwareVersionsNotCurrent);
+        //
+        // for (final
+        // org.opensmartgridplatform.domain.core.valueobjects.FirmwareVersion
+        // firmwareVersion : versionsNotCurrent) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)